### ISSUE TITLE
Mark single word as untranslatable outside the pipeline

### ISSUE THREAD

**Issue by unhammer** [apertium/apertium#33]
It'd be nice to be able to mark a single word (or short phrase) as untranslatable outside the pipeline, but without turning it into a superblank like `<apertium-notrans>` does. 

This way, CG and apertium-tagger could notice it and treat it e.g. as a proper noun or whatever makes sense, and transfer rules could be properly blocked by it, while people using apertium could preprocess such that e.g. `STARTUP` in `STARTUP is Making the World a Better Place` shouldn't be translated to `OPPSTART`.

**Comment by khannatanmai** [comment]
@unhammer  Secondary tags could help here.

**Comment by TinoDidriksen** [comment]
Yup. All sorts of outside information could be passed along this way.

**Comment by xavivars** [comment]
I think the key point here is _outside the pipeline_. How would you add secondary tags to a word outside of the pipeline? I guess that would require also changes to lt-proc to understand that new "do-not-translate-this-word" markup into a secondary tag....

**Comment by khannatanmai** [comment]
@xavivars I know it says outside the pipeline, but as I understand it, whether to translate or not translate a word is something that will be computed in the pipeline. Unless if you want to manually provide a list of words that shouldn't be translated. (Isn't this something that can be done in t1x already? Give a list and if the word is part of that list then propagate the SL lemma. It won't be generated but that's fine.)

Or when you say outside the pipeline, do you mean actually having a markup in the input corpus? If you mean that analysis of the word should produce the LU with a <don't translate> tag, that can certainly be done in the monodix if needed.

**Comment by xavivars** [comment]
Yes, I mean completely outside the pipeline. Apertium currently supports doing that "from outside" the pipeline. You can send this text

    <apertium-notrans>This text will not be translated</apertium-notrans>, but this one will.

and the "html" parser will convert everything `<apertium-notrans>` as superblanks.

**Comment by TinoDidriksen** [comment]
> and the "html" parser will convert everything `<apertium-notrans>` as superblanks...

...which is the problem. That ruins contexts. It'd be much better to let it have an analysis that makes sense in context.

Something like `<apertium-notrans pos="np">STARTUP</apertium-notrans>` which would get turned into `^STARTUP/STARTUP<np><m:notrans>$` by something and respected by lt-proc, or done by lt-proc directly.

Yes, it would require changes to lt-proc, but that will be needed anyway for things like markup handling. Might as well make that part generic also.

**Comment by xavivars** [comment]
Definitely, not saying we can't touch lt-proc. My point was as a reply to this 

> @xavivars I know it says outside the pipeline, but as I understand it, whether to translate or not translate a word is something that will be computed in the pipeline.

Unhammer's request was being able to do that from outside the pipeline.

**Comment by unhammer** [comment]
Yes, Xavi interpreted my request correctly :) This was for users who don't want to / are not able to change the translator at all, but just need a way to make something in their texts untranslatable.

**Comment by khannatanmai** [comment]
Ah alright I thought the request was about a more general "Mark words that one shouldn't translate". But my initial comment was about secondary tags helping and as @TinoDidriksen said, the <notrans> markup can be converted to a secondary tag (the same way we would convert html tags to secondary tags and attach them to word LUs.

**Comment by ftyers** [comment]
Note that this is also related to the issue of codeswitching, e.g. identifying and marking spans as not translatable because they are in another language. I put some thoughts [here](https://wiki.apertium.org/wiki/User-based_chunking).
